src/field_theory/krull_topology.lean

```diff
@@ -287,3 +287,222 @@ begin
 end
 
 end totally_disconnected
+
+section compact
+
+open set
+
+
+universe u
+
+/-- Let `E` be an intermediate field of `L/K` with `E/K` finite, and let `f` be an ultrafilter
+  on `L ≃ₐ[K] L`. Then the restriction map `(L ≃ₐ[K] L) → (E →ₐ[K] L)` pushes `f` forward
+  to an ultrafilter on `E →ₐ[K] L`. Since `E →ₐ[K] L` is a finite set, this ultrafilter is
+  principal. The element of `E →ₐ[K] L` generating this principal ultrafilter is
+  `alg_hom_of_finite_dimensional_of_ultrafilter h_findim f`, where
+  `h_findim : finite_dimensional K E`. -/
+noncomputable def alg_hom_of_finite_dimensional_of_ultrafilter
+  {K : Type*} {L : Type u} [field K] [field L] [algebra K L]
+  {E : intermediate_field K L} (h_findim : finite_dimensional K E)
+  (f : ultrafilter (L ≃ₐ[K] L)) : E →ₐ[K] L :=
+classical.some
+  (@ultrafilter.eq_principal_of_fintype (E →ₐ[K] L) _
+  (f.map (λ σ, σ.to_alg_hom.comp (intermediate_field.val E))))
+
+/-- Let `f` be an ultrafilter on `L ≃ₐ[K] L`. For an intermediate field `E` of `L/K`, there is a
+  natural restriction map `res : (L ≃ₐ[K] L) → (E →ₐ[K] L)`. Moreover, this restriction map gives
+  a pushforward of `f` to an ultrafilter on `E →ₐ[K] L`. If `E/K` is finite, then this pushforward
+  is generated by `alg_hom_of_finite_dimensional_of_ultrafilter h_findim f`.  -/
+lemma alg_hom_of_finite_dimensional_of_ultrafilter_spec {K L : Type*} [field K] [field L]
+  [algebra K L] {E : intermediate_field K L} (h_findim : finite_dimensional K E)
+  (f : ultrafilter (L ≃ₐ[K] L)) : (f.map (λ σ : L ≃ₐ[K] L, σ.to_alg_hom.comp
```

Comment:
Why are you developing the API in terms of `ultrafilter (L ≃ₐ[K] L)` rather than `ultrafilter (L →ₐ[K] L)`? The latter would be strictly more general, but maybe you do not want to carry around a `map alg_equiv.to_alg_hom`.

Reply:
Good idea! Have done it:)